data/World/Overworld.json

```diff
@@ -1385,8 +1385,31 @@
     {
         "region_name": "Market Treasure Chest Game",
         "scene": "Market Treasure Chest Game",
+        "hint": "MARKET",
         "locations": {
-            "Market Treasure Chest Game Reward": "can_use(Lens_of_Truth)"
+            "Market Treasure Chest Game Salesman": "is_child and at_night",
+            "Market Treasure Chest Game Reward": "(shuffle_tcgkeys == 'vanilla' and can_use(Lens_of_Truth)) or (shuffle_tcgkeys != 'vanilla' and
```

Comment:
These can be massively simplified.

* You don't need to check for keysy, that's handled by logically giving the keys as starting items.
* You don't need to check for vanilla other than to see if lens is required. Let the logic assume that you can open both chests, which ends up being equivalent.

For example, I'd write the logic for the reward as:

```python
(shuffle_tcgkeys != 'vanilla' or can_use(Lens_of_Truth)) and (Small_Key_Treasure_Chest_Game, 6)
```

Reply:
addressed